pygeoapi/api.py

```diff
@@ -1429,6 +1434,57 @@ def get_collection_items(
                 HTTPStatus.INTERNAL_SERVER_ERROR, headers, request.format,
                 'NoApplicableCode', msg)
 
+        crs_transform_wkt = None
```

Comment:
The lines 1437-1487 and 2189-2238 are identical, or almost. Please make a utility function. #1155 introduces a function `util.get_supported_crs_list` that may be called within at later (merging) stage.

Reply:
Yes, I have mentioned this redundancy in the **Additional Information** section of the PR's description. The lines covering the generation of the `supported_crs_list` is already covered by https://github.com/geopython/pygeoapi/pull/1155 's function `util.get_supported_crs_list`. In addition, it does need to be done twice (processing the `bbox-crs` parameter already needs it). I thought about remove it when merging master into https://github.com/geopython/pygeoapi/pull/1149.

For the rest however, I am not sure what's best. A utility function should take as parameter a `APIRequest` object, the `provider_def` `dict`, the `API` object itself (for error handling, when raising exceptions with `self.get_exception()`), the `headers` object (both for the `self.get_exception()` call and for setting `Content-Crs`). This utility function should return a `CrsTransformWkt` instance, or `None` if no transformation should be done.

I feel like many of these parameters are very specific to both the `API` class and the `APIRequest` to pass them to a function of the generic module `util.py`. I thought that having a dedicated method the `API` class would be a good solution (removes duplication and the need to pass the `API` instance as parameter, `self`), but then it would break the current design of the `API` class: one method for one end-point, or at least the method should be very generic and usable by many other methods, not just specific to two methods and only related to OGC API - Features...

Maybe a helper function in the same `api.py` module would be a better fit? There are already such functions (e.g. `validate_bbox`, `validate_datetime`...).

Reply:
Just pushed some changes to make both `get_collection_items` and `get_collection_item` more compact. I used your approach with two new `@staticmethod`s. I have also re-used your `get_supported_crs_list` util function from https://github.com/geopython/pygeoapi/pull/1155. Does this correspond to what you had in mind?

Reply:
Yes, good!